include/array/array.h

```diff
@@ -2264,12 +2264,39 @@ class array {
    * `alloc`. This is a deep copy of the contents of `other`. */
   array(const array& other, const Alloc& alloc) : array(alloc) { assign(other); }
 
+  /** Move construct from another array `other`. If the allocator of this array
+   * is propagate on container move assignment, this operation moves the allocation
+   * of other to this array, and the other array becomes a default constructed
+   * array. If the allocator of this and the other array are non-equal, each
+   * element is move-constructed into a new allocation. */
+  array(array&& other) : buffer_(nullptr), buffer_size_(0), base_(nullptr) {
+    using std::swap;
+    if (std::allocator_traits<allocator_type>::propagate_on_container_move_assignment::value) {
+      swap(alloc_, other.alloc_);
+      swap(buffer_, other.buffer_);
+      swap(buffer_size_, other.buffer_size_);
+      swap(base_, other.base_);
+      swap(shape_, other.shape_);
+    } else {
+      alloc_ = Alloc();
```

Comment:
I think you should remove this, it's already happening by default. This is constructing and assigning an extra allocator.

Reply:
Oops you are right. Removed.